File: 16255/16255.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    16255.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/04/22 15:43:56 by jihoolee          #+#    #+#              #
#    Updated: 2022/04/22 16:05:33 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_min_ballcount(param):
    k = param[0]
    bigger = max(param[1:])
    smaller = min(param[1:])

    if (bigger >= k):
        return 2 - (bigger - smaller)
    elif (smaller + 2 >= k):
        return smaller + 2 - bigger
    else:
        return k - bigger

n_cases = int(input())
games_lst = [list(map(int, input().split())) for _ in range(n_cases)]
results = list(map(get_min_ballcount, games_lst))

for result in results:
    print(result)

File: 16255/16255.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   16255.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/04/22 15:05:16 by jihoolee          #+#    #+#             */
/*   Updated: 2022/04/22 15:56:46 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>

int max(int x, int y) {
  if (x > y)
    return x;
  return y;
}

int get_min_ballcount(int k, int x, int y) {
  if (x >= k)
    return 2 - (x - y);
  else if (y + 2 >= k)
    return y + 2 - x;
  else
    return k - x;
}

int main(void) {
  int n_cases;
  int k, x, y;

  std::cin >> n_cases;
  for (int i = 0; i < n_cases; ++i) {
    std::cin >> k >> x >> y;
    std::cout << get_min_ballcount(k, max(x, y), x + y - max(x, y)) << std::endl;
  }
  return 0;
}
